src/bindings/interface/issue.i

```diff
@@ -0,0 +1,123 @@
+
+%feature("docstring") libcellml::Issue::setComponent
+"Sets the :class:`Component` that this issue is relevant to (``None`` to unset).";
+
+%feature("docstring") libcellml::Issue::importSource
+"Returns the :class:`ImportSource` that this issue is relevant to (or ``None``).";
+
+%feature("docstring") libcellml::Issue::setImportSource
+"Sets the :class:`ImportSource` that this issue is relevant to (``None`` to unset).";
+
+%feature("docstring") libcellml::Issue::model
+"Returns the :class:`Model` that this issue is relevant to (or ``None``).";
+
+%feature("docstring") libcellml::Issue::setModel
+"Sets the :class:`Model` that this issue is relevant to (``None`` to unset).";
+
+%feature("docstring") libcellml::Issue::units
+"Get the :class:`Units` that this issue is relevant to (or ``None``).";
+
+%feature("docstring") libcellml::Issue::setUnits
+"Sets the :class`Units` that this issue is relevant to (``None`` to unset).";
+
+%feature("docstring") libcellml::Issue::variable
+"Get the :class:`Variable` that this issue is relevant to (or ``None``).";
+
+%feature("docstring") libcellml::Issue::setVariable
+"Sets the :class:`Variable` that this issue is relevant to (``None`` to unset).";
+
+%feature("docstring") libcellml::Issue::reset
+"Get the :class:`Reset` that this issue is relevant to (or ``None``).";
+
+%feature("docstring") libcellml::Issue::setReset
+"Sets the :class:`Reset` that this issue is relevant to (``None`` to unset).";
+
+%{
+#include "libcellml/issue.h"
+%}
+
+%create_constructor(Issue)
+%extend libcellml::Issue {
+    Issue(const ComponentPtr &component) {
+        auto ptr = new std::shared_ptr<  libcellml::Issue >(libcellml::Issue::create(component));
+        return reinterpret_cast<libcellml::Issue *>(ptr);
+    }
+    Issue(const ImportSourcePtr &importSource) {
+        auto ptr = new std::shared_ptr<  libcellml::Issue >(libcellml::Issue::create(importSource));
+        return reinterpret_cast<libcellml::Issue *>(ptr);
+    }
+    Issue(const ModelPtr &model) {
+        auto ptr = new std::shared_ptr<  libcellml::Issue >(libcellml::Issue::create(model));
+        return reinterpret_cast<libcellml::Issue *>(ptr);
+    }
+    Issue(const ResetPtr &reset) {
+        auto ptr = new std::shared_ptr<  libcellml::Issue >(libcellml::Issue::create(reset));
+        return reinterpret_cast<libcellml::Issue *>(ptr);
+    }
+    Issue(const UnitsPtr &units) {
+        auto ptr = new std::shared_ptr<  libcellml::Issue >(libcellml::Issue::create(units));
+        return reinterpret_cast<libcellml::Issue *>(ptr);
+    }
+    Issue(const VariablePtr &variable) {
+        auto ptr = new std::shared_ptr<  libcellml::Issue >(libcellml::Issue::create(variable));
+        return reinterpret_cast<libcellml::Issue *>(ptr);
+    }
+}
+
+%include "libcellml/exportdefinitions.h"
+%include "libcellml/referencerules.h"
+%include "libcellml/types.h"
+%include "libcellml/issue.h"
```

Comment:
Reorder the headers.

Reply:
Done.

Reply:
Undone!  Changing the order makes the build fail ... ?

Reply:
Hmm, what error message are you getting?

Reply:
[ 98%] Building CXX object src/bindings/python/CMakeFiles/issue.dir/swig_intermediaries/issuePYTHON_wrap.cxx.o
/Users/kmoy001/libcellml/libcellml-build/src/bindings/python/swig_intermediaries/issuePYTHON_wrap.cxx:3157:58: error: use of undeclared identifier 'ComponentPtr'
SWIGINTERN libcellml::Issue *new_libcellml_Issue__SWIG_1(ComponentPtr const &component){
                                                         ^
/Users/kmoy001/libcellml/libcellml-build/src/bindings/python/swig_intermediaries/issuePYTHON_wrap.cxx:3157:88: error: expected ';' after top level declarator
SWIGINTERN libcellml::Issue *new_libcellml_Issue__SWIG_1(ComponentPtr const &component){
                                                                                       ^
                                                                                       ;
2 errors generated.
make[2]: *** [src/bindings/python/CMakeFiles/issue.dir/swig_intermediaries/issuePYTHON_wrap.cxx.o] Error 1
make[1]: *** [src/bindings/python/CMakeFiles/issue.dir/all] Error 2
make: *** [all] Error 2

Reply:
In every single other file, the file's interface counterpart is included last on the list.  Resolving this now ...

Reply:
> In every single other file, the file's header counterpart is included last on the list. Resolving this now ...

Indeed, I can see that now. Hmm... I guess there must some technical reasons for it. @hsorby, @MichaelClerx?